faroe/api/src/main/java/org/n52/faroe/controller/SettingDefinitionsController.java

```diff
@@ -0,0 +1,138 @@
+        }
+    }
+
+    /**
+     *
+     */
+    @PutMapping(value = "/{id}")
+    public ResponseEntity<Object> updateSettingDefinition(@PathVariable("service") String serviceName,
+                                                          @PathVariable("id") String definitionId,
+                                                          @RequestBody String body) {
+        LOG.info(String.format("Updating definition for service %s", serviceName));
+        try {
+            final Service service = servicesDao.getServiceByName(serviceName);
+            final JsonObject settingDefinitionRequestBody = new JsonParser().parse(body)
+                    .getAsJsonObject();
+            final JsonObject settingDefinitionGroupRequestBody = settingDefinitionRequestBody
+                    .getAsJsonObject("SettingDefinitionGroup");
+            final SettingDefinition<?> requiredSettingDefinition = service.getSettingsService()
+                    .getDefinitionByKey(definitionId);
+            final SettingDefinitionGroup requiredSettingDefinitionGroup = requiredSettingDefinition
+                    .getGroup();
+            Optional.ofNullable(settingDefinitionRequestBody.get(Constants.DESCRIPTION)).ifPresent(
+                    description -> requiredSettingDefinition.setDescription(description.getAsString()));
+            Optional.ofNullable(settingDefinitionRequestBody.get(Constants.TITLE))
+                    .ifPresent(title -> requiredSettingDefinition.setTitle(title.getAsString()));
+            Optional.ofNullable(settingDefinitionRequestBody.get(Constants.OPTIONAL))
+                    .ifPresent(optional -> requiredSettingDefinition.setOptional(optional.getAsBoolean()));
+            Optional.ofNullable(settingDefinitionRequestBody.get(Constants.KEY))
+                    .ifPresent(key -> requiredSettingDefinition.setKey(key.getAsString()));
+            Optional.ofNullable(settingDefinitionRequestBody.get(Constants.ORDER))
+                    .ifPresent(order -> requiredSettingDefinition.setOrder(order.getAsFloat()));
+            Optional.ofNullable(settingDefinitionGroupRequestBody.get(Constants.TITLE))
+                    .ifPresent(title -> requiredSettingDefinitionGroup.setTitle(title.getAsString()));
+            Optional.ofNullable(settingDefinitionGroupRequestBody.get(Constants.DESCRIPTION)).ifPresent(
+                    description -> requiredSettingDefinitionGroup.setDescription(description.getAsString()));
+            requiredSettingDefinition.setGroup(requiredSettingDefinitionGroup);
+            return new ResponseEntity<>(requiredSettingDefinition, HttpStatus.OK);
+        } catch (Exception e) {
+            LOG.error(String.format("Couldn't update setting definition %s for service %s",
+                                definitionId, serviceName));
+            e.printStackTrace();
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
```

Comment:
I think it's better to create an exception for that and include an exception mapper for the exception than always to return a `ResponseEntity`.